test/katex-spec.js

```diff
@@ -2956,6 +2986,12 @@ describe("Newlines via \\\\ and \\newline", function() {
     it("should not allow \\cr at top level", () => {
         expect("hello \\cr world").toNotBuild();
     });
+
+    it("array redefines and resets \\\\", () => {
+        expect("a\\\\b\\begin{matrix}x&y\\\\z&w\\end{matrix}\\\\c")
+            .toParseLike("a\\newline b\\begin{matrix}x&y\\cr z&w\\end{matrix}" +
+                         "\\newline c");
+    });
 });
```

Comment:
@kevinbarabash It works in LaTeX.

Reply:
Yes, that will work too. However, as another commented-out test mentions, `\global\mydef...` doesn't work yet (but should), because I'm not grabbing the argument to `\global` correctly.

Reply:
@edemaine did you changes address this?

Reply:
No.  I'll add some TODOs.